src/main/java/com/exasol/containers/workarounds/WorkaroundManager.java

```diff
@@ -10,44 +10,59 @@
 public class WorkaroundManager {
     private static final Logger LOGGER = LoggerFactory.getLogger(WorkaroundManager.class);
     private final Workaround[] workarounds;
+    private final Set<String> previouslyAppliedWorkarounds;
 
     /**
      * Create a new instance of a {@link WorkaroundManager} with a list of workarounds it is responsible for.
      *
-     * @param workarounds list of workarounds the manager needs to check and apply
+     * @param previouslyAppliedWorkarounds set of workarounds that have already been applied
+     * @param workarounds                  list of workarounds the manager needs to check and apply
      * @return new workaround manager
      */
-    public static WorkaroundManager create(final Workaround... workarounds) {
-        return new WorkaroundManager(workarounds);
+    public static WorkaroundManager create(final Set<String> previouslyAppliedWorkarounds,
+            final Workaround... workarounds) {
+        return new WorkaroundManager(previouslyAppliedWorkarounds, workarounds);
     }
 
-    private WorkaroundManager(final Workaround[] workarounds) {
+    private WorkaroundManager(final Set<String> previouslyAppliedWorkarounds, final Workaround[] workarounds) {
+        this.previouslyAppliedWorkarounds = previouslyAppliedWorkarounds;
         this.workarounds = workarounds;
     }
 
     /**
      * Apply workarounds.
      *
+     * @return list of applied workarounds
+     *
      * @throws WorkaroundException if any workaround could not be applied.
      */
-    public void applyWorkarounds() throws WorkaroundException {
-        final List<Workaround> appliedWorkarounds = new ArrayList<>();
+    // [impl->dsn~workaround-manager-applies-multiple-of-workarounds~1]]
+    public Set<Workaround> applyWorkarounds() throws WorkaroundException {
+        final Set<Workaround> appliedWorkarounds = new HashSet<>();
         for (final Workaround workaround : this.workarounds) {
             applyWorkaround(workaround);
             appliedWorkarounds.add(workaround);
         }
         if (!appliedWorkarounds.isEmpty() && LOGGER.isInfoEnabled()) {
             LOGGER.info("Applied workarounds: {}", joinWorkaroundNames(appliedWorkarounds));
```

Comment:
I thin kit would make more sense to only show this if the workaround was really applied. Applyed sounds like it was needed. Maybe return a boolen from applyWorkarround?

Reply:
There is indeed an `if` missing.